src/mongocxx/result/bulk_write.hpp

```diff
@@ -71,18 +74,11 @@ class MONGOCXX_API bulk_write {
     std::int32_t upserted_count() const;
 
     ///
-    /// Gets the ids of the inserted documents.
-    ///
-    /// @return The values of the _id field for inserted documents.
-    ///
-    bsoncxx::document::element inserted_ids() const;
-
-    ///
     /// Gets the ids of the upserted documents.
     ///
-    /// @return The values of the _id field for upserted documents.
+    /// @return A map from bulk write index to _id field for upserted documents.
```

Comment:
Since the elements in the values in the map are views into the document owned by the bulk write, if the bulk write object is destroyed, the map is invalidated. You should add a comment stating that the returned id_map must not be accessed after the bulk_write object is destroyed.